### PR TITLE
add versioneer

### LINKED ISSUE #35: Add a distribution package version
...  oops.

### PR DESCRIPTION
**Pull request by gordon-epc**
Add versioneer code to satisfy #35

### CODE REVIEW

**Comment by altendky** [comment]
Along with excluding from black, we can exclude the version file from coverage as well.

https://coverage.readthedocs.io/en/coverage-5.4/source.html#execution

Looks like we would use `[run] omit` in the `.coveragerc`.  Oh, hmm, both black and coverage should skip the `versioneer.py` file as well.

At least some of this could be avoided with a [PEP 518](https://www.python.org/dev/peps/pep-0518/)-compatible version of versioneer.  There is one and there was [discussion about merging that functionality](https://github.com/python-versioneer/python-versioneer/issues/193).  I wouldn't switch to [versioneer-518](https://pypi.org/project/versioneer-518/) now, just setting context for another bit to be done...  someday.  [PEP 518](https://www.python.org/dev/peps/pep-0518/) is about describing build-time dependencies.  It basically deprecates `setuptools.setup(setup_requires=)`, plus allows a project to specify use of something other than setuptools entirely.  I think that [ciborg](https://github.com/altendky/ciborg/blob/ce792b745058b8068679b63cc0e314c2a2787667/pyproject.toml#L4) may be the only project where I used [versioneer-518](https://pypi.org/project/versioneer-518/).

**Comment by altendky** [comment]
So, we started missing branch coverage on the `if` that was a workaround for an unexplored oddity from pytest.  There hasn't been a recent release of pytest so that isn't the cause.  If this were hard to 'fix' (or turns out to be), I would forgo mandating it be addressed prior to merging this PR.  But, I think we can take care of it simply.  Just open a new PR adding a `# pragma: no branch` to that `if` and let's see if it is happy.  If you want to get fancy, add a ticket for that `TODO` comment (and add a reference to the ticket in the code).

https://codecov.io/gh/altendky/ssst/src/8a736ecd8648475f3dccb868be5bed065621d5f7/src/ssst/_tests/test_cli.py
```python
# TODO: pytest seems to make a duplicate entry compared to checking with
#  'just Python'.  No idea why right now, just accepting it.
our_consolidated_console_scripts = []
for script in our_console_scripts:
    if script not in our_consolidated_console_scripts:
        our_consolidated_console_scripts.append(script)

assert len(our_consolidated_console_scripts) == 1, our_consolidated_console_scripts
```

File: setup.cfg
Comment (altendky) [inline setup.cfg:104]:
```suggestion
tag_prefix = 'v'
```
(similar to https://github.com/epcpower/sunspec-demo/blob/81be80cb68bf0019a1478c3693e0c5b7883ab648/setup.cfg#L47)

**Review by altendky — CHANGES_REQUESTED** [review]
Since we aren't about to make a release or anything, I think I'll hold off on an approval and merge until we get the build stuff worked out.  Either waiting for Codecov, or getting impatient and setting up the (hopeful) workaround.

File: setup.cfg
Comment (altendky) [reply setup.cfg:104]: \*bump\* here.  Also, hopefully the build will pass with an update merge from `main` (optionally using the button down this page near the merge button).

Comment (gordon-epc) [reply setup.cfg:104]: Did we reach a conclusion on the choice of using a tag prefix of `v` vs `release/` ? That's why I was holding on this change request.

Comment (altendky) [reply setup.cfg:104]: Sorry, I totally forgot that whole conversation.  I restarted in slack and will summarize here when we finish it.

Comment (altendky) [reply setup.cfg:104]: RTD offers no apparent control and seems to 'expect' a bare version number based on their `v: <exact tag>` combo box for version selection.  Since I have no predicted other use for tags...  whatever.  Leave it as `''` because or switch to `'v'` for consistency.  No big deal.

Comment (gordon-epc) [reply setup.cfg:104]: My conclusion is to leave as is, with no tag_prefix. Versions will be in the format `1.2.3`.

**Review by altendky — APPROVED** [review]

**Pull request merged**
